test/functional/test_framework/test_node.py

```diff
@@ -625,6 +625,10 @@ def add_outbound_p2p_connection(self, p2p_conn, *, wait_for_verack=True, p2p_idx
 
         This method adds the p2p connection to the self.p2ps list and returns
         the connection to the caller.
+
+        p2p_idx must be different for simultaneously connected peers. When reusing it for the next peer
+        after disconnecting the previous one, it is necessary to wait for the disconnect to finish to avoid
+        a race condition.
```

Comment:
Maybe mention that `disconnect_p2ps` avoids the race?

Reply:
I tried to describe it in more general terms because `disconnect_p2ps` can only be used if there is no other, unrelated connection that we need to keep.